### PR TITLE
Commenting of graphic dialogs' main components relationships

### PR DESCRIPTION
**Pull request by FrancoisJRenaud**
Made some comments and asked some questions and set some tasks within the code. Don't know how orthodox you find this practice. I am trying to get this code well commented in order to smoothen our work and in the future enable the open source quality of the product to be fully appreciated. For the moment exploration and understanding is really painful. Hopefully it will not be as painful for the next newbie... These in-line comments are mainly related to this project.

### CODE REVIEW

**Comment by FrancoisJRenaud** [comment]
One additional question that I would like to discuss is the suspected absence of plot options in dlgGeneralForGraphics. How do we deal with all what's available on sdgPlotOptions (title, themes, ...) ? Do we wish to add a plot option button ? If yes does it lead to our sdgPlotOptions ? What about the fact we deal with Layers there ? Both sdgPlotOptions and dlgGeneralForGraphics have a ucrAdditionalLayers. Could these be cloned perhaps when opening sdgPLotOptions and then when returning in the other direction ? What about the RSyntax ... On the user's side doesn't look good to be able to add layers in the plot options whereas it's the purpose of the main dialog...

**Comment by FrancoisJRenaud** [comment]
The last commit (implementation of debug 3 of issue #1931) included a significant amount of changes in plenty of files. Would be good to merge before conflicts arise...

File: instat/UcrGeomListWithAes.vb [hunk 1]
```diff
@@ -61,20 +62,23 @@ Public Class UcrGeomListWithParameters
...
         End If
-        UcrSelector.SetDataframe(strGlobalDataFrame, (Not bUseGlobalAes) OrElse strGlobalDataFrame = "")
+        UcrSelector.SetDataframe(strGlobalDataFrame, (Not bApplyAesGlobally) OrElse strGlobalDataFrame = "")
         UcrSelector.Reset()
         bCurrentFixAes = bFixAes
         SetAes(bCurrentFixAes)
-        chkApplyOnAllLayers.Checked = bUseGlobalAes
+        'Using the values of the two relevant parameters, the two following lines determine whether the chkBoxes ApplyToAllLayers and IgnoreGlobalAes should be ticked. 
+        chkApplyOnAllLayers.Checked = bApplyAesGlobally
+        chkIgnoreGlobalAes.Checked = bIgnoreGlobalAes
```
Comment (dannyparsons) [inline instat/UcrGeomListWithAes.vb]: You need to be a bit careful here. If you look at the checked changed events of these controls. For example, you can't have both checked, but that is possible here. I think you just need to check that case here before setting the checked values.

File: instat/sdgLayerOptions.vb [hunk 2]
```diff
@@ -56,19 +59,20 @@ Public Class sdgLayerOptions
     End Sub
 
     Public Sub SetRSyntax(clsRSyntax As RSyntax)
-
+        'Question to discuss: sdgLayerOptions has no RSyntax, what would this function be used for ? Should it be deleted ? It is called in dlgBoxplot and dlgDotPlot but obviously doesn't perform anything...
```
Comment (dannyparsons) [inline instat/sdgLayerOptions.vb]: Delete if doesn't do anything. Probably not needed if everything is done through RFunctions

File: instat/ucrAdditionalLayers.vb [hunk 3]
```diff
@@ -123,14 +129,26 @@ Public Class ucrAdditionalLayers
...
         If clsSelectedGeom.GetParameter("mapping") IsNot Nothing Then
             clsLocalAes = clsSelectedGeom.GetParameter("mapping").clsArgumentFunction
         Else
             clsLocalAes = Nothing
         End If
-        sdgLayerOptions.SetupLayer(clsTempGgPlot:=dlgGeneralForGraphics.clsRggplotFunction, clsTempGeomFunc:=clsSelectedGeom, clsTempAesFunc:=dlgGeneralForGraphics.clsGgplotAesFunction, bFixAes:=False, bFixGeom:=True, strDataframe:=strGlobalDataFrame, bUseGlobalAes:=False, clsTempLocalAes:=clsLocalAes)
+
+        'Before we set-up the Layer in sdgLayerOptions, we determine the value of bIgnoreGlobalAes. We can detect if chkIgnoreGlobalAes was ticked last time the Layer was editted by looking at the "inherit.aes" parameter of the layer parameters.
+        clsTempRParameter = clsSelectedGeom.GetParameter("inherit.aes")
+        If (Not clsTempRParameter Is Nothing) AndAlso (clsTempRParameter.strArgumentValue = "False") Then
```
Comment (dannyparsons) [inline instat/ucrAdditionalLayers.vb]: I normally use `IsNot Nothing`. I'm not sure if there is any reason to use one or the other so probably fine.

**Comment by FrancoisJRenaud** [comment]
Agreed for the change parameters, although why would we keep apply on all layers true and not ignore global aes ? The reason for this bug can be anything, so I was thinking better to have everything blank rather than an incoherent tick.

**Comment by dannyparsons** [comment]
Possibly, but having one checked is closer to what was input in the first place. But as this is a bigger problem if this happens anyway, I don't feel that strongly.

**Comment by FrancoisJRenaud** [comment]
Sure, but we don't know what was input for that layer in the first place as you said. For sure it was not both ticks as this is impossible on the dialog. this would come from an internal error, also I think that choosing one tick randomly is more risky, either you get it right or completely wrong. Whereas when you don't tick anything, you only ever do one mistake.

**Comment by dannyparsons** [comment]
I'm happy with that as well

**Comment by FrancoisJRenaud** [comment]
:) am I eligible for merge then ?

**Pull request merged**